File: rust/src/models/component_parameterization.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::{component_parameters::ComponentParameters, price_component_id::PriceComponentId};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ComponentParameterization {
    pub component_id: PriceComponentId,

    pub parameters: ComponentParameters,
}

impl ComponentParameterization {
    pub fn new(component_id: PriceComponentId, parameters: ComponentParameters) -> Self {
        Self {
            component_id,
            parameters,
        }
    }
}

File: typescript/src/models/componentParameterization.ts
// this file is @generated
import {
  type ComponentParameters,
  ComponentParametersSerializer,
} from "./componentParameters";
import { type PriceComponentId, PriceComponentIdSerializer } from "./priceComponentId";

export interface ComponentParameterization {
  componentId: PriceComponentId;

  parameters: ComponentParameters;
}

export const ComponentParameterizationSerializer = {
  _fromJsonObject(object: any): ComponentParameterization {
    return {
      componentId: PriceComponentIdSerializer._fromJsonObject(object["component_id"]),
      parameters: ComponentParametersSerializer._fromJsonObject(object["parameters"]),
    };
  },

  _toJsonObject(self: ComponentParameterization): any {
    return {
      component_id: PriceComponentIdSerializer._toJsonObject(self.componentId),
      parameters: ComponentParametersSerializer._toJsonObject(self.parameters),
    };
  },
};
